amino.go

```diff
@@ -2,21 +2,29 @@ package amino
 
 import (
 	"bytes"
-	"encoding/binary"
-	"encoding/json"
-	"errors"
 	"fmt"
 	"io"
 	"reflect"
 	"time"
+
+	"encoding/binary"
+	"encoding/json"
+
+	"github.com/pkg/errors"
 )
 
-//----------------------------------------
-// Global methods for global sealed codec.
-var gcdc *Codec
+var (
+	// Global methods for global sealed codec.
+	gcdc *Codec
+
+	// we use this time to init. a zero value (opposed to reflect.Zero which gives time.Time{} / 01-01-01 00:00:00)
+	zeroTime time.Time
 
-// we use this time to init. a zero value (opposed to reflect.Zero which gives time.Time{} / 01-01-01 00:00:00)
-var zeroTime time.Time
+	// NotPointerErr is thrown when you call a method that expects a pointer, e.g. Unmarshal
+	NotPointerErr = errors.New("expected a pointer")
```

Comment:
I'm not familiar with stdlib's `errors` vs. `pkg/errors` vs. `fmt.Errorf`, but I saw this dependency isn't used elsewhere. Worth adding a new library for?

Reply:
pkg/errors allows you to wrap the error with some context instead of just passing it on. This helps callers to understand exactly where the error occurred. Although you can achieve sth. similar with fmt.Errorf, I think error.Wrap is slightly more convenient (and it also adds the call stack if you want to).

Reply:
This is an excellent read on errors and touches the different error packages too: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully